app/src/Model/Table/ArticlesTable.php

```diff
@@ -8,19 +8,29 @@
 // the EventInterface class
 use Cake\Event\EventInterface;
 
-// add this use statement right below the namespace declaration to import
 // the Validator class
 use Cake\Validation\Validator;
 
+// the Query class
+use Cake\ORM\Query;
+
 class ArticlesTable extends Table
 {
     public function initialize(array $config): void
     {
         $this->addBehavior('Timestamp');
+        // relations
+        $this->belongsToMany('Tags', [
+            'joinTable' => 'articles_tags',
+            'dependent' => true
+        ]);
```

Comment:
It's an interesting topic for you to understand further about CakePHP's ORM Association https://book.cakephp.org/4/en/orm/associations.html

Please check later.